incubator/hnc/pkg/controllers/config_controller_test.go

```diff
@@ -0,0 +1,37 @@
+package controllers_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("HNCConfiguration", func() {
+	ctx := context.Background()
+
+	var (
+		fooName string
+		barName string
+	)
+
+	BeforeEach(func() {
+		fooName = createNS(ctx, "foo")
+		barName = createNS(ctx, "bar")
+	})
+
+	It("should create a new ObjectReconciler when adding a new type to the HNCConfiguration", func() {
+		setParent(ctx, barName, fooName)
+		makeSecret(ctx, fooName, "foo-sec")
+
+		// "foo-sec" is not propagated to bar because Secret hasn't been configured in HNCConfiguration.
+		Eventually(hasSecret(ctx, barName, "foo-sec")).Should(BeFalse())
```

Comment:
Probably better to add more confirmation here by either:
1) add a sleep time before this to give it a "chance" to propagate object; OR
2) add another case for comparison: when creating an object in the `foo` NS with a type configured in HNCConfiguration, e.g. `configmap`,  the parent `configmap` object is propagated to the `bar` NS while the `secret` object is not. It will look like:
```
config := getHNCConfig(ctx)
// Omit this, if the configmap is already configured in the original singleton.
addConfigmapToHNCConfig(ctx, config)

setParent(ctx, barName, fooName)
makeSecret(ctx, fooName, "foo-sec")
makeConfigmap(ctx, fooName, "foo-cfg")

Eventually(hasConfigmap(ctx, barName, "foo-cfg")).Should(BeTrue())
Eventually(hasSecret(ctx, barName, "foo-sec")).Should(BeFalse())
```

Reply:
Thanks so much for the suggestion! I used option 1 for now to let it sleep for 1 second to have a chance to propagate.

I will add another test case later to compared the propagation of an object with type that is configured in the HNCConfiguration with another object with type that is not configured. We can do that by compare any type with RBAC types on start up because we should have configured RBAC types by default. Right now this part is tricky since the code does not configure RBAC types by default and we set it to default in suite_test.go. After I correct the code and remove the default setup in suite_test.go, I will add that test case.

Reply:
sgtm, yes option 2 will be added much easier once default singleton is configured or you have more cases to test later.

Reply:
I agree, option #1 seems simpler and also sufficient in this case.

Reply:
Is 1 second long enough? Wont normal flakiness of a running test system potentially take longer complete a reconcile? If I'm reading this correctly we wait 1 second to ensure HNC didn't accidentally propagate something it shouldn't.

Reply:
In practice, if things propagate in the test, they usually propagate in well under 1s. And yes, we're trying to ensure that HNC _hasn't_ done something that it _shouldn't_, so even if this only "works" 80% of the time, that should be more than enough to inform the developer if they accidentally break this.

Reply:
My concern is more around the point (months from now) when it starts taking longer than 1s and it only starts to work <10% of the time. Realistically now we don't have any indication if the test is actually successfully testing what it should. I think option 2 is the overall better test, and I assume we'll implement it eventually? I'm not concerned enough to hold up the merge.

Reply:
Thanks Adrian and Ryan for the suggestions. In this case, I will keep the option 1 in this PR and switch to Option 2 in my next PR when I fix the default config setup as I discussed in the previous comment.

Reply:
Ok, sounds reasonable, since Sophie is going to be adding a special-purpose test for the default configuration.